src/webqc-eri.c

```diff
@@ -0,0 +1,332 @@
+        if ( (*eri_index_a)[i] != (*eri_index_b)[i] ) {
+            rv = false;
+        }
+    }
+    return rv;
+}
+
+
+static int
+eri_index_to_memory_position(WQC *handler, const eri_index_t *eri_index)
+{
+    unsigned int n = handler->eri_info.number_of_functions;
+    unsigned int offset = 1;
+    unsigned int pos = 0;
+
+    for ( unsigned int i = 0 ; i < 4 ; i++ ) {
+        pos += offset * (*eri_index)[3-i];
+        offset*=n;
+    }
+
+    return pos;
+}
+
+static bool
+eri_available_in_handler(WQC *handler, const eri_index_t *eri_index)
+{
+    unsigned int first_available_function = eri_index_to_memory_position(handler, &handler->eri_info.eri_values.begin_eri_index);
+    unsigned int end_available_function = eri_index_to_memory_position(handler, &handler->eri_info.eri_values.end_eri_index);
+    unsigned int idx_position = eri_index_to_memory_position(handler, eri_index);
+    return idx_position >= first_available_function && idx_position < end_available_function;
+}
+
+bool wqc_next_eri_index(
+    WQC *handler,
+    eri_index_t *eri_index
+)
+{
+    unsigned int number_of_functions = handler->eri_info.number_of_functions;
+    (*eri_index)[3]++;
+    if ( (*eri_index)[3] == number_of_functions ) {
+        (*eri_index)[3] = 0;
+        (*eri_index)[2]++;
+        if ( (*eri_index)[2] == number_of_functions ) {
+            (*eri_index)[2] = 0;
+            (*eri_index)[1]++;
+            if ((*eri_index)[1] == number_of_functions) {
+                (*eri_index)[1] = 0;
+                (*eri_index)[0]++;
+            }
+        }
+    }
+
+    return eri_available_in_handler(handler, eri_index);
+}
+
+
+bool
+wqc_get_eri_value(  WQC *handler,  const eri_index_t *eri_index, double *eri_value, double *eri_precision)
+{
+    bool rv = false;
+
+    if ( eri_available_in_handler(handler, eri_index )) {
+        unsigned int idx_position = eri_index_to_memory_position(handler, eri_index);
+        unsigned int first_available_function = eri_index_to_memory_position(handler, &handler->eri_info.eri_values.begin_eri_index);
+
+        *eri_precision = handler->eri_info.eri_values.eri_precision;
+        *eri_value = handler->eri_info.eri_values.eri_values[idx_position-first_available_function];
+        rv = true;
+    }  else {
+        wqc_set_error_with_message(handler, WEBQC_NOT_FETCHED , "Reading ERI value that was not retrieved from the WQC server");
+    }
+
+    return rv;
+}
+
+
+static void
+func_index_to_shell_index(WQC *handler, const eri_index_t *func_range, unsigned int *shell_index)
+{
+    for ( unsigned int i = 0U ; i < 4 ; ++i ) {
+        if ((*func_range)[i] == handler->eri_info.number_of_functions) {
+            shell_index[i] = handler->eri_info.number_of_shells;
+        } else {
+            shell_index[i] = handler->eri_info.basis_functions[(*func_range)[i]].shell_index;
+        }
+    }
+}
+
+
+bool
+wqc_fetch_ERI_values(WQC *handler, const eri_index_t *eri_range_begin, const eri_index_t *eri_range_end)
```

Comment:
Too many lines. Mostly because of the `snprintf`

Reply:
slip to two.